src/utils/iconutils.cpp

```diff
@@ -79,21 +79,34 @@ QString IconUtils::replaceForegroundOfIcon(const QString &p_iconContent, const Q
 
 bool IconUtils::isMonochrome(const QString &p_iconContent)
 {
-    // Match color-hex codes.
-    QRegExp monoRe("#([0-9a-fA-F]{6}|[0-9a-fA-F]{3})");
-
+    // Match all color-hex codes.
+    QRegExp hexRe("#([0-9a-fA-F]{6}|[0-9a-fA-F]{3})");
     int i = 0;
-    QMap<QString, int> fill_color;
-
-    while ((i = monoRe.indexIn(p_iconContent, i)) != -1) {
-        fill_color[monoRe.cap(1)] = 0;
-        i += monoRe.matchedLength();
+    QStringList fillColor;
```

Comment:
Why need to store all the matches? let last = first, if last != current, then abort.